File: bms/go/go-edlib/find/find.go
package test

var ()

const ()

func find(slice [][]rune, val []rune) int {
	for i, item := range slice {
		if Equal(item, val) {
			return i
		}
	}
	return -1
}

func Equal(a, b []rune) bool {
	if len(a) != len(b) {
		return false
	}
	for i, v := range a {
		if v != b[i] {
			return false
		}
	}
	return true
}

File: translations_only/fail/go/go-edlib/find/claude2/Hinted/find.rs

use std::cmp::Ordering;

fn find(slice: &[Box<[isize]>], val: &[isize]) -> isize {
    for (i, item) in slice.iter().enumerate() {
        if equal(item, val) {
            return i as isize;
        }
    }
    -1
}

fn equal(a: &[isize], b: &[isize]) -> bool {
    if a.len() != b.len() {
        return false;
    }
    for i in 0..a.len() {
        match a[i].cmp(&b[i]) {
            Ordering::Equal => continue,
            _ => return false,
        }
    }
    true
}

